Lesson19/script.js

```diff
@@ -79,48 +108,71 @@ class ShoppingCart {
   }
 
   // In real life, we usually use id to find elements or do any manipulations
-  removeItem(name) {
-    for (let i = 0; i < this.#items.length; i++) {
-      const currentElement = this.#items[i];
-      if (name.toLowerCase() === currentElement.name.toLowerCase()) {
-        this.#items.splice(i, 1);
-        return;
-      }
+  removeItem(id) {
+    console.log("Removing item from the cart...");
+    const itemtoRemove = this.#items.find((item) => item.id === id);
+    if (!itemtoRemove) {
+      console.log("There is no item with this id");
+      this.viewCart();
+      return;
+    }
+    itemtoRemove.quantity--; //find methoduyla istediğimiz özellikteki objectin kendisini(adresni) bulduk
+
+    if (itemtoRemove.quantity > 0) {
+      console.log(
+        `Quantity reduced: ID ${itemtoRemove.id} | ${itemtoRemove.name} miktarı 1 azaltıldı. Yeni miktar: ${itemtoRemove.quantity}`,
+      );
+    } else if (itemtoRemove.quantity === 0) {
+      console.log(
+        `Item removed: ID ${itemtoRemove.id} | ${itemtoRemove.name} sepetten çıkarıldı.`,
+      );
+      this.#items = this.#items.filter((item) => item.id !== id); //filter methodunda id si eşleşmeyenleri yani silinmeyenleri diiznin yeni haline aktar.
+      this.viewCart();
+    }
+  }
+  getTotal() {
+    let total = 0;
+    for (const item of this.#items) {
+      total += item.price * item.quantity;
+    }
+    console.log(`The total price is ${total}`);
+    return total;
+  }
+  applyDiscount(discountCode) {
+    const discountAmount = [
+      { code: "SAVE10", discount: 0.1 },
+      { code: "SAVE20", discount: 0.2 },
+      { code: "SAVE30", discount: 0.3 },
+      { code: "SAVE50", discount: 0.5 },
+    ];
+    const checkDiscount = discountAmount.find(
+      (item) => item.code === discountCode,
+    );
+    if (!checkDiscount) {
+      console.warn(`Please log valid discount code`);
+      return;
```

Comment:
`return this.getTotal()` would be better choice as there will be a value returned in both cases.